packages/dapp/src/components/DashboardDisplay/CredentialTable/index.tsx

```diff
@@ -0,0 +1,380 @@
+import React, { useMemo, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { QueryCredentialsRequestResult } from '@blockchain-lab-um/masca-connector';
+import { CheckCircleIcon, XCircleIcon } from '@heroicons/react/24/outline';
+import { EllipsisVerticalIcon } from '@heroicons/react/24/solid';
+import {
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownMenu,
+  DropdownTrigger,
+  Pagination,
+  Selection,
+  SortDescriptor,
+  Table,
+  TableBody,
+  TableCell,
+  TableColumn,
+  TableHeader,
+  TableRow,
+  Tooltip,
+} from '@nextui-org/react';
+import { VerifiableCredential } from '@veramo/core';
+import { encodeBase64url } from '@veramo/utils';
+import { useTranslations } from 'next-intl';
+
+import DeleteModal from '@/components/DeleteModal';
+import { ShareCredentialModal } from '@/components/ShareCredentialModal';
+import StoreIcon from '@/components/StoreIcon';
+import { convertTypes } from '@/utils/string';
+import { useTableStore } from '@/stores';
+import { LastFetched } from '../LastFetched';
+import { sortCredentialList } from '../utils';
+
+type Key = string | number;
+type SelectedKeys = 'all' | Iterable<Key> | undefined;
+
+const ROWS_PER_PAGE = 3;
+
+const columns = [
+  {
+    key: 'type',
+    label: 'TYPE',
+    allowSorting: true,
+  },
+  {
+    key: 'date',
+    label: 'DATE',
+    allowSorting: true,
+  },
+  {
+    key: 'subject',
+    label: 'SUBJECT',
+    allowSorting: true,
+  },
+  {
+    key: 'issuer',
+    label: 'ISSUER',
+    allowSorting: true,
+  },
+  {
+    key: 'exp_date',
+    label: 'EXPIRATION DATE',
+    allowSorting: true,
+  },
+  {
+    key: 'status',
+    label: 'STATUS',
+  },
+  {
+    key: 'data_store',
+    label: 'DATA STORE',
+  },
+  {
+    key: 'actions',
+    label: 'ACTIONS',
+  },
+];
+
+interface CredentialTableProps {
+  vcs: QueryCredentialsRequestResult[];
+}
+
+const CredentialTable = ({ vcs }: CredentialTableProps) => {
+  const t = useTranslations('Dashboard');
+  const [selectedKeys, setSelectedKeys] = useState<SelectedKeys>(new Set([]));
+  const [sortDescriptor, setSortDescriptor] = useState<SortDescriptor>();
+  const [deleteModalOpen, setDeleteModalOpen] = useState(false);
+  const [shareModalOpen, setShareModalOpen] = useState(false);
+  const [selectedVC, setSelectedVC] = useState<QueryCredentialsRequestResult>();
+  const [shareCredential, setShareCredential] =
+    useState<VerifiableCredential>();
+  const [page, setPage] = useState(1);
+
+  const router = useRouter();
+
+  const { setSelectedVCs } = useTableStore((state) => ({
+    setSelectedVCs: state.setSelectedVCs,
+  }));
+
+  const renderCell = React.useCallback(
+    (vc: QueryCredentialsRequestResult, columnKey: React.Key) => {
+      switch (columnKey) {
+        case 'type':
+          // eslint-disable-next-line no-case-declarations
+          const types = convertTypes(vc.data.type);
+          return <span className="font-bold">{types.split(',')[0]}</span>;
```

Comment:
```suggestion
          return <span className="font-bold">{convertTypes(vc.data.type).split(',')[0]}</span>;
```

Better to do it this way. Or you can also change the `const  types` to `let types` and move it outside of the switch statement.